src/library_openal.js

```diff
@@ -314,6 +314,21 @@ var LibraryOpenAL = {
       }
       {{{ makeSetValue('data', '0', '0', 'i32') }}};
       break;
+    case 0x1007 /* ALC_FREQUENCY */:
+      if (!device) {
+        AL.alcErr = 0xA001 /* ALC_INVALID_DEVICE */;
+        return 0;
+      }
+      {{{ makeSetValue('data', '0', '(new (window.AudioContext || window.webkitAudioContext)).sampleRate', 'i32') }}};
```

Comment:
spec is vague indeed.
when in doubt, I look at [openal-soft](https://github.com/kcat/openal-soft) implementation, the de-facto standard.
and it returns the most precise ALC_FREQUENCY only when current context is set.

so yes, it makes sense to require AL.currentContext here to obtain the information.